### PR TITLE
Add ability to parse pub props in msg CB

### PR DESCRIPTION
**Pull request by embhorn**
This is a feature request from ZD13146

### CODE REVIEW

File: src/mqtt_client.c [hunk 1]
```diff
@@ -278,7 +278,8 @@ static int MqttClient_RespList_Find(MqttClient *client,
 #endif /* WOLFMQTT_MULTITHREAD */
 
 #ifdef WOLFMQTT_V5
-static int Handle_Props(MqttClient* client, MqttProp* props, byte use_cb)
+static int Handle_Props(MqttClient* client, MqttProp* props, byte use_cb,
+                        byte free)
```
Comment (dgarske) [inline src/mqtt_client.c:282]: `free` might be a reserved keyword. Consider different variable name.

**Review by dgarske — CHANGES_REQUESTED** [review]
Seems like a reasonable way to solve this. We do have a property callback, but I can understand wanting to access the properties in the publish callback as well.

File: src/mqtt_client.c (hunk 1 above)
Comment (embhorn) [reply src/mqtt_client.c]: Check

**Comment by embhorn** [comment]
Let me try to get feedback from the customer before merging

**Review by dgarske — APPROVED** [review]

File: src/mqtt_client.c [hunk 2]
```diff
@@ -892,6 +902,7 @@ static int MqttClient_WaitType(MqttClient *client, void *packet_obj,
             else {
                 /* use generic packet object */
                 use_packet_obj = &client->msg;
+                XMEMSET(use_packet_obj, 0, sizeof(MqttObject));
```
Comment (embhorn) [inline src/mqtt_client.c:905]: @dgarske   This fixes several fuzzer issues. There was stale data left from processing other incoming messages.

Comment (dgarske) [reply src/mqtt_client.c:905]: Would prefer `sizeof(client->msg)`. Also the read lock is held here, which is good, but consider also wrapping in `rc = wm_SemLock(&client->lockClient);` and `wm_SemUnlock(&client->lockClient);`.

**Review by dgarske — CHANGES_REQUESTED** [review]

**Review by dgarske — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Review by embhorn — COMMENTED** [review]
Excellent. Fix for property handling checks out.

**Comment by dgarske** [comment]
Customer confirmed fix. Eric also approved my changes and I approved him. Merging for both of us.

**Pull request merged**